modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/credential/CredentialCenter.java

```diff
@@ -75,7 +83,23 @@ boolean deleteCredentialByType(Csp csp, String userName, CredentialType credenti
      * @param userName The userName to get.
      */
     public AbstractCredentialInfo getCredential(Csp csp, String userName) {
-        return null;
+        Map<String, AbstractCredentialInfo> userCredentialMap = credentialInfoMap.get(csp);
+        if (userCredentialMap != null) {
+            AbstractCredentialInfo credentialInfo = userCredentialMap.get(userName);
+            if (credentialInfo != null) {
+                return credentialInfo;
+            }
+        }
+        AbstractCredentialInfo credentialInfo = findCredentialInfoFromEnv(csp);
+        if (credentialInfo == null) {
+            throw new IllegalStateException(
+                    String.format(
+                            "No credential information found for the given Csp:%s and user:%s.",
+                            csp, userName));
+        }
+        userCredentialMap = credentialInfoMap.computeIfAbsent(csp, k -> new HashMap<>());
```

Comment:
Not need to store the credential that form the Env for every userName.

Reply:
ok,I optimized him.